### PR TITLE
allows to set hostname

### PR DESCRIPTION
**Pull request by javierwilson**
this allows to set the hostname when "hostname" is defined , and domain name if specified

### CODE REVIEW

File: ansible/roles/common/tasks/main.yml [hunk 1]
```diff
@@ -60,3 +60,22 @@
...
+  when: hostname is defined
+  tags:
+    - hostname
+
+- name: "Sets /etc/hosts"
+  become: yes
+  lineinfile:
+    dest: /etc/hosts
+    regexp: '^{{ ansible_default_ipv4.address }} '
+    line: '{{ ansible_default_ipv4.address }} {{ hostname }}.{{ internal_domain_name }} {{ hostname }}'
```
Comment (sravfeyn) [inline ansible/roles/common/tasks/main.yml]: Nice!

File: ansible/roles/common/tasks/main.yml [hunk 2]
```diff
@@ -60,3 +60,22 @@
     line: "session required pam_limits.so"
   tags:
     - nofile_limits
+
+
+- name: "Sets hostname"
+  become: yes
+  hostname:
+    name: "{{ hostname }}"
```
Comment (sravfeyn) [inline ansible/roles/common/tasks/main.yml]: Where is this defined? Some of already set hostnames on these servers might not match with what you have set in the DNS. May be double check this before deploying?

File: ansible/vars/icds/icds_public.yml [hunk 3]
```diff
@@ -2,6 +2,7 @@ SITE_HOST: 'www.icds-cas.gov.in'
 CAS_SITE_HOST: 'cas.commcarehq.org'
 NO_WWW_SITE_HOST: 'icds-cas.gov.in'
 deploy_env: icds
+internal_domain_name: 'internal-icds.commcarehq.org'
```
Comment (benrudolph) [inline ansible/vars/icds/icds_public.yml]: may also want to set this for the rest of the envs

File: ansible/roles/common/tasks/main.yml (hunk 2 above)
Comment (javierwilson) [inline ansible/roles/common/tasks/main.yml]: I do this at the hostname inventory level like  127.0.0.1 hostname="localhost"

File: ansible/roles/common/tasks/main.yml (hunk 1 above)
Comment (sravfeyn) [inline ansible/roles/common/tasks/main.yml]: There's `[etc_hosts](https://github.com/dimagi/commcarehq-ansible/blob/master/ansible/roles/common/tasks/main.yml#L5)` tags that does similar editing already. Should we merge these two?

**Pull request merged**